torchmetrics/metric.py

```diff
@@ -377,7 +378,19 @@ def wrapped_func(*args: Any, **kwargs: Any) -> None:
             self._computed = None
             self._update_count += 1
             with torch.set_grad_enabled(self._enable_grad):
-                update(*args, **kwargs)
+                try:
+                    update(*args, **kwargs)
+                except RuntimeError as e:
+                    if "Expected all tensors to be on" in str(e):
+                        traceback.print_exc()
+                        raise RuntimeError(
+                            f"{str(e)}. \n"
+                            "This could be due to the metric class not being on the same device as input.\n"
+                            "Instead of `metric=Metric()` try to do `metric=Metric().to(device)` where"
+                            " device corresponds to the device of the input."
+                        )
+                    raise e
```

Comment:
```suggestion
                try:
                    update(*args, **kwargs)
                except RuntimeError as err:
                    if "Expected all tensors to be on" in str(err):
                        traceback.print_exc()
                        raise RuntimeError(
                            f"{str(err)}. \n"
                            "This could be due to the metric class not being on the same device as input.\n"
                            "Instead of `metric=Metric()` try to do `metric=Metric().to(device)` where"
                            " device corresponds to the device of the input."
                        )
                    raise err
```
single letter var is tricky... :)